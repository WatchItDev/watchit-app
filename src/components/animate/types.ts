// ----------------------------------------------------------------------

type EaseType =
  | "linear"
  | "easeIn"
  | "easeOut"
  | "easeInOut"
  | "circIn"
  | "circOut"
  | "circInOut"
  | "backIn"
  | "backOut"
  | "backInOut"
  | "anticipate"
  | number[];

export interface VariantsType {
  distance?: number;
  durationIn?: number;
  durationOut?: number;
  easeIn?: EaseType;
  easeOut?: EaseType;
}

export interface TranHoverType {
  duration?: number;
  ease?: EaseType;
}

export interface TranEnterType {
  durationIn?: number;
  easeIn?: EaseType;
}

export interface TranExitType {
  durationOut?: number;
  easeOut?: EaseType;
}

export interface BackgroundType {
  colors?: string[];
  duration?: number;
  ease?: EaseType;
}
