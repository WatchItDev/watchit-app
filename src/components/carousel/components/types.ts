import {BoxProps} from "@mui/material/Box";
import {SxProps, Theme} from "@mui/material/styles";
import {IconifyProps} from "@src/components/iconify";
import {StackProps} from "@mui/material/Stack";
import React from "react";
import {IconButtonProps} from "@mui/material/IconButton";

export interface CarouselDotsProps extends BoxProps {
  rounded?: boolean;
  sx?: SxProps<Theme>;
}

export type ArrowIconProps = {
  icon?: IconifyProps; // Right icon
  isRTL?: boolean;
};

export type CarouselArrowIndexProps = {
  index: number;
  total: number;
  icon?: IconifyProps; // Right icon
  onNext?: VoidFunction;
  onPrev?: VoidFunction;
  sx?: SxProps<Theme>;
};

export interface CarouselArrowsProps extends StackProps {
  shape?: 'circular' | 'rounded';
  filled?: boolean;
  children?: React.ReactNode;
  icon?: IconifyProps; // Right icon
  onNext?: VoidFunction;
  onPrev?: VoidFunction;
  leftButtonProps?: IconButtonProps;
  rightButtonProps?: IconButtonProps;
}

export interface CarouselArrowsStyledIconButtonProps extends IconButtonProps {
  filled?: boolean;
  shape?: 'circular' | 'rounded';
  hasChild?: boolean;
}

export interface NavigationArrowsProps {
  next: () => void;
  prev: () => void;
}

export type CarouselDotsStyledRootProps = {
  rounded: boolean;
};

