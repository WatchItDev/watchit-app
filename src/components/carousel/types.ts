import React from "react";

import {Profile, VideoMetadataV3} from "@lens-protocol/api-bindings";
// @ts-ignore
import { Post,PublicationMetadataMediaVideoFieldPolicy } from "@lens-protocol/api-bindings/dist/declarations/src/lens/graphql/generated";

import {StackProps} from "@mui/material/Stack";
import {IconButtonProps} from "@mui/material/IconButton";

import {BoxProps} from "@mui/material/Box";
import {SxProps, Theme} from "@mui/material/styles";
import {IconifyProps} from "@src/components/iconify";


export interface CarouselDotsProps extends BoxProps {
  rounded?: boolean;
  sx?: SxProps<Theme>;
}

interface CarouselNavigationActions {
  onNext?: VoidFunction;
  onPrev?: VoidFunction;
}

export type ArrowIconProps = {
  icon?: IconifyProps;
  isRTL?: boolean;
};

export type CarouselArrowIndexProps = CarouselNavigationActions & {
  index: number;
  total: number;
  icon?: IconifyProps;
  sx?: SxProps<Theme>;
};

export interface CarouselArrowsProps extends StackProps, CarouselNavigationActions {
  shape?: 'circular' | 'rounded';
  filled?: boolean;
  children?: React.ReactNode;
  icon?: IconifyProps; // Right icon
  leftButtonProps?: IconButtonProps;
  rightButtonProps?: IconButtonProps;
}

export interface CarouselArrowsStyledIconButtonProps extends IconButtonProps {
  filled?: boolean;
  shape?: 'circular' | 'rounded';
  hasChild?: boolean;
}

export interface NavigationArrowsProps {
  next: () => void;
  prev: () => void;
}

export type CarouselDotsStyledRootProps = {
  rounded: boolean;
};

interface CarouselResizable {
  minItemWidth: number;
  maxItemWidth: number;
}

export interface CarouselWrapperProps<T> extends CarouselResizable{
  data: T[];
  title?: string;
  renderSlide: (slideItems: T[], itemsPerRow: number, index: number) => React.ReactNode;
  carouselSettings: any;
  boxStyle?: any;
  boxClassName?: string;
}

export interface CarouselSlideProps<T> {
  items: T[];
  itemsPerRow: number;
  renderItem: (item: T) => React.ReactNode;
}

export type CarouselCreatorsProps = CarouselResizable & {
  data: Profile[];
  title?: string;
};

export type CarouselPosterMiniProps = CarouselResizable & {
  data: Post[];
  title?: string;
};

export type CarouselTopTitlesProps = {
  posts: Post[];
  category?: string;
};

export type CarouselPosterSlideProps = {
  items: Post[];
  itemsPerRow: number;
};

export interface PublicationType extends Post {
  altTag: string;
  metadata: VideoMetadataV3 | PublicationMetadataMediaVideoFieldPolicy;
}
