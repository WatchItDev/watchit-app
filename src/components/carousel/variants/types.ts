import {Profile} from "@lens-protocol/api-bindings";
// @ts-ignore
import type {Post} from "@lens-protocol/api-bindings/dist/declarations/src/lens/graphql/generated";
import React from "react";

export interface CarouselWrapperProps<T> {
  data: T[];
  title?: string;
  minItemWidth: number;
  maxItemWidth: number;
  renderSlide: (slideItems: T[], itemsPerRow: number, index: number) => React.ReactNode;
  carouselSettings: any;
  boxStyle?: any;
  boxClassName?: string;
}

export type CarouselSlideProps = {
  items: Profile[];
  itemsPerRow: number;
};

export type CarouselCreatorsProps = {
  data: Profile[];
  title?: string;
  minItemWidth: number;
  maxItemWidth: number;
};

export type CarouselPosterMiniProps = {
  data: Post[];
  title?: string;
  minItemWidth: number;
  maxItemWidth: number;
};

export type CarouselTopTitlesProps = {
  posts: Post[];
  category?: string;
};

export type CarouselPosterSlideProps = {
  items: Post[];
  itemsPerRow: number;
};
