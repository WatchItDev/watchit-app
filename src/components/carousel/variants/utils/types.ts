export interface calculateItemsPerSlideProps {
  parentWidth: number;
  minItemWidth: number;
  maxItemWidth: number;
}
