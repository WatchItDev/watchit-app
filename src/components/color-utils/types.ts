// @mui
import { StackProps } from '@mui/material/Stack';

// ----------------------------------------------------------------------

export interface ColorPreviewProps extends StackProps {
  colors: string[];
  limit?: number;
}

export interface ColorPickerProps extends StackProps {
  multi?: boolean;
  colors: string[];
  selected: string | string[];
  limit?: 'auto' | number;
  onSelectColor: (color: string | string[]) => void;
}
