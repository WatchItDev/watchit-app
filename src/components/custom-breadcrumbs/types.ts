// @mui
import { BreadcrumbsProps } from '@mui/material/Breadcrumbs';

// ----------------------------------------------------------------------

export type BreadcrumbsLinkProps = {
  name?: string;
  href?: string;
  icon?: React.ReactElement;
};

export interface CustomBreadcrumbsProps extends BreadcrumbsProps {
  heading?: string;
  moreLink?: string[];
  activeLast?: boolean;
  action?: React.ReactNode;
  links: BreadcrumbsLinkProps[];
}
