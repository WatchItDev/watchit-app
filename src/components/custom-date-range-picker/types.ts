// ----------------------------------------------------------------------

export type DateRangePickerProps = {
  startDate: Date | null;
  endDate: Date | null;
  onChangeStartDate: (newValue: Date | null) => void;
  onChangeEndDate: (newValue: Date | null) => void;
  //
  open: boolean;
  onOpen?: VoidFunction;
  onClose: VoidFunction;
  onReset?: VoidFunction;
  //
  selected?: boolean;
  error?: boolean;
  //
  label?: string;
  shortLabel?: string;
  //
  title?: string;
  variant?: 'calendar' | 'input';
  //
  setStartDate?: React.Dispatch<React.SetStateAction<Date | null>>;
  setEndDate?: React.Dispatch<React.SetStateAction<Date | null>>;
};
