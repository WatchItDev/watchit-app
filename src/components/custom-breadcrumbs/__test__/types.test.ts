import React from 'react'
import { BreadcrumbsLinkProps, CustomBreadcrumbsProps } from '../types';

describe('[TYPES]: Breadcrumbs types testing', () => {
  it('to match snapshot', () => {
    const sampleLink: BreadcrumbsLinkProps = {
      name: 'Dashboard',
      href: '/dashboard',
      icon: React.createElement('span', null, '🏠'),
    };

    const sampleBreadcrumbs: CustomBreadcrumbsProps = {
      heading: 'Page Title',
      links: [
        { name: 'Home', href: '/' },
        { name: 'Dashboard', href: '/dashboard' },
      ],
      activeLast: true,
      moreLink: ['https://example.com'],
      action: React.createElement('button', null, 'Action'),
    };

    expect({ sampleLink, sampleBreadcrumbs }).toMatchSnapshot();
  });

  it('creates valid BreadcrumbsLinkProps with all properties', () => {
    const link: BreadcrumbsLinkProps = {
      name: 'Settings',
      href: '/settings',
      icon: React.createElement('span', null, '⚙️'),
    };

    expect(link).toHaveProperty('name', 'Settings');
    expect(link).toHaveProperty('href', '/settings');
    expect(link).toHaveProperty('icon');
  });

  it('creates valid BreadcrumbsLinkProps with only required properties', () => {
    // All properties are optional in BreadcrumbsLinkProps
    const minimalLink: BreadcrumbsLinkProps = {};
    expect(minimalLink).toEqual({});
  });

  it('creates valid CustomBreadcrumbsProps with all properties', () => {
    const breadcrumbs: CustomBreadcrumbsProps = {
      heading: 'Users',
      links: [
        { name: 'Home', href: '/' },
        { name: 'Users', href: '/users' },
      ],
      activeLast: true,
      moreLink: ['https://docs.example.com/users'],
      action: React.createElement('button', null, 'Add user'),
      sx: { marginBottom: 2 }
    };

    expect(breadcrumbs).toHaveProperty('heading', 'Users');
    expect(breadcrumbs).toHaveProperty('links');
    expect(breadcrumbs.links).toHaveLength(2);
    expect(breadcrumbs).toHaveProperty('activeLast', true);
    expect(breadcrumbs).toHaveProperty('moreLink');
    expect(breadcrumbs).toHaveProperty('action');
    expect(breadcrumbs).toHaveProperty('sx');
  });

  it('creates valid CustomBreadcrumbsProps with only required properties', () => {
    const minimalBreadcrumbs: CustomBreadcrumbsProps = {
      links: [{ name: 'Home', href: '/' }],
    };

    expect(minimalBreadcrumbs).toHaveProperty('links');
    expect(minimalBreadcrumbs.links).toHaveLength(1);
    expect(minimalBreadcrumbs.heading).toBeUndefined();
    expect(minimalBreadcrumbs.activeLast).toBeUndefined();
    expect(minimalBreadcrumbs.moreLink).toBeUndefined();
    expect(minimalBreadcrumbs.action).toBeUndefined();
  });

  it('supports an array of strings for moreLink property', () => {
    const breadcrumbs: CustomBreadcrumbsProps = {
      links: [{ name: 'Home' }],
      moreLink: ['https://example.com/docs', 'https://example.com/help'],
    };

    expect(Array.isArray(breadcrumbs.moreLink)).toBe(true);
    expect(breadcrumbs.moreLink).toHaveLength(2);
    expect(typeof breadcrumbs.moreLink![0]).toBe('string');
  });

  it('supports empty links array', () => {
    const breadcrumbs: CustomBreadcrumbsProps = {
      links: [],
    };

    expect(breadcrumbs.links).toEqual([]);
  });
});
