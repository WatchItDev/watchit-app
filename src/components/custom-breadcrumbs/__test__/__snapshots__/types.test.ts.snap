// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[TYPES]: Breadcrumbs types testing > to match snapshot 1`] = `
{
  "sampleBreadcrumbs": {
    "action": <button>
      Action
    </button>,
    "activeLast": true,
    "heading": "Page Title",
    "links": [
      {
        "href": "/",
        "name": "Home",
      },
      {
        "href": "/dashboard",
        "name": "Dashboard",
      },
    ],
    "moreLink": [
      "https://example.com",
    ],
  },
  "sampleLink": {
    "href": "/dashboard",
    "icon": <span>
      üè†
    </span>,
    "name": "Dashboard",
  },
}
`;
